spark/src/main/scala/org/apache/spark/sql/delta/DeltaParquetFileFormat.scala

```diff
@@ -166,17 +177,23 @@ case class DeltaParquetFileFormat(
       }
       results.headOption.map(e => ColumnMetadata(e._2, e._1))
     }
-    val isRowDeletedColumn = findColumn(IS_ROW_DELETED_COLUMN_NAME)
-    val rowIndexColumn = findColumn(ROW_INDEX_COLUMN_NAME)
 
-    if (isRowDeletedColumn.isEmpty && rowIndexColumn.isEmpty) {
-      return parquetDataReader // no additional metadata is needed.
+    val isRowDeletedColumn = findColumn(IS_ROW_DELETED_COLUMN_NAME)
+    val rowIndexColumnName = if (useMetadataRowIndex) {
+      ParquetFileFormat.ROW_INDEX_TEMPORARY_COLUMN_NAME
     } else {
-      // verify the file splitting and filter pushdown are disabled. The new additional
-      // metadata columns cannot be generated with file splitting and filter pushdowns
-      require(!isSplittable, "Cannot generate row index related metadata with file splitting")
-      require(disablePushDowns, "Cannot generate row index related metadata with filter pushdown")
+      ROW_INDEX_COLUMN_NAME
```

Comment:
nit: may be rename this to `INTERNAL_ROW_INDEX_COLUMN_NAME`. It's not a strong preference, but it may make the code easier to understand.

Reply:
I will keep it like this for now. We should remove this code path soon-ish.